### DISCUSSION TITLE
Default loadbalancer for nginx ingress also targets control plane nodes

### DISCUSSION THREAD

**Discussion by MariusSchiffer** [kube-hetzner/terraform-hcloud-kube-hetzner#446] [Q&A]
I have created a fresh cluster with 5 agent nodes and 3 control plane nodes.  
Unfortunately, the ingress loadbalancer not only targets the agent nodes, but also the control plane nodes.

Do you have any idea what influences this?

**Comment by mysticaltech** [comment]
@MariusSchiffer If using Hetzner LB, it shouldn't be case, unless you set `allow_scheduling_on_control_plane` to true, in which case you need to set it to false.

Now if you use klipperLB instead of the Hetzner one, then it's something expected. For an HA setup like yours, best to use the Hetzner LB.

**Reply by MariusSchiffer** [reply]
`allow_scheduling_on_control_plane` is set to false and I'm using the Hetzner LB

**Reply by mysticaltech** [reply]
Ok, in that case, please share your `kube.tf`, without sensitive values and with comments removed. I will have a look at it ASAP.

**Reply by MariusSchiffer** [reply]
I also used a fresh kube.tf.example and modified it only for agent nodepools, SSH key+token, disabled traefik/enabled nginx, cluster name

**Reply by mysticaltech** [reply]
Ok, turning this into an issue, will investigate further ASAP this week. But in the meantime, Traefik should not have this problem.

**Reply by MariusSchiffer** [reply]
Here you go: http://ix.io/4hZ3

**Reply by mysticaltech** [reply]
@MariusSchiffer If it can help, I have a good feeling that this is caused by the default Helm config we give to Nginx. And this can be changed at the bottom of the `kube.tf`, if you want to read the ingress-nginx helm file doc and play with this, maybe you can make it work it, if so please let us know.

**Reply by MariusSchiffer** [reply]
Strangely enough, this also happens with traefik. I did upgrade the terraform provider beforehand

**Reply by ifeulner** [reply]
Hm, nginx is providing it's service via ClusterIP, so isn't that by default available on all nodes? And therefore the load balancer addresses also all nodes? (servers and agents)?

**Reply by mysticaltech** [reply]
@ifeulner Before at least the LB did not take control planes as targets unless we allowed scheduling on the control plane. However, I will double-check later, to see if I can reproduce the issue.

**Reply by mysticaltech** [reply]
FYI folks, https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/447#issuecomment-1350597300
